src/compiler/parser.ts

```diff
@@ -5605,6 +5605,10 @@ namespace ts {
                 }
 
                 if (isPropertyAccess) {
+                    // TODO: issue a better error?
+                    if (expression.kind === SyntaxKind.ExpressionWithTypeArguments) {

```

Comment:
This might not be right; I think I need to do a lookahead parse above instead so this doesn't return having consumed the `.` or `?.`.